### PR TITLE
Iss75

### PR DESCRIPTION
**Pull request by HomoCodens**

### CODE REVIEW

**Comment by HomoCodens** [comment]
We should keep c25cead and 4caa71c anyway in case the user enters a `date_format` that isn't sorted correctly (although that would probably bring its own problems).

File: R/utils.R [hunk 1]
```diff
@@ -147,7 +147,7 @@ formatNumericDate <- function(date, freq, date_format = NULL) {
   } else {
     month <- floor(12*(date - year + 1/24)) + 1  # Why "+ 1/24"? Because floating point arithmetic. 12*0.0833333 may not be 12.
     if(is.null(date_format)) {
-      return(sprintf("%d-%d", year, month))
+      return(sprintf("%d-%02d", year, month))
```
Comment (mbannert) [inline R/utils.R]: Does the leading zero already help the sorting?

File: R/write_ts.R [hunk 2]
```diff
@@ -99,19 +99,21 @@ write_ts <- function(tl,
...
         if("index" %in% names(tsdf)) {
           tsdf <- tsdf[, -"index", with = FALSE]
         }
         
-        setcolorder(tsdf, c(nTs+1, seq(nTs)))
-        setnames(tsdf, c("date", names(tl)))
-        
         if(transpose) {
-          tsdf <- dcast(melt(tsdf, id.vars = "date", variable.name = "series"), series ~ date)
+          tsdf <- dcast(melt(tsdf, id.vars = "t", variable.name = "series"), series ~ t)
```
Comment (mbannert) [inline R/write_ts.R]: ist this data.table's dcast being used here?

File: R/utils.R (hunk 1 above)
Comment (HomoCodens) [reply R/utils.R]: Yes.
```
> sort(c("2018-3", "2018-2", "2018-10"))
[1] "2018-10" "2018-2"  "2018-3" 
> sort(c("2018-03", "2018-02", "2018-10"))
[1] "2018-02" "2018-03" "2018-10"
```

File: R/write_ts.R (hunk 2 above)
Comment (HomoCodens) [reply R/write_ts.R]: We only import `data.table` so I would say yes. Looks like `reshape2`s is only used in the tests.

**Pull request merged**